src/backend/distributed/commands/table.c

```diff
@@ -2301,9 +2311,41 @@ PreprocessAlterTableSchemaStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
-	ErrorIfTenantTable(relationId, "ALTER TABLE SET SCHEMA");
-	ErrorIfTenantSchema(get_namespace_oid(stmt->newschema, false),
-						"ALTER TABLE SET SCHEMA");
+	/* Undistribute table if it is under a tenant schema */
+	Oid oldSchemaId = get_rel_namespace(relationId);
```

Comment:
We should not allow undistribution of the table if any table (only tenant tables in the old schema can refer to it) is referencing it via a foreign key.

Reply:
We now check that before undistributing tenant table.